Add a parameter to pass tokenizer to llama QNN runner

limintang opened the pull request:
Differential Revision: D77910880

cccclai commented on examples/qualcomm/oss_scripts/llama/runner/runner.cpp:
```diff
@@ -115,30 +117,36 @@ Error Runner::load() {
-        "failed to load tokenizer %s",
-        tokenizer_path_.c_str());
+  if (tokenizer_ != nullptr) {
+    eos_ids->insert(200008);
```
What is this 200008? It seems very model specific.

limintang replied:
> What is this 200008? It seems very model specific.

Replied in phabricator.

limintang commented:
@pytorchbot label "release notes: none"

cccclai reviewed (approved):
Thanks for addressing the comments!

limintang committed:
Add a parameter to pass tokenizer to llama QNN runner (#12285)

Summary: Pull Request resolved: https://github.com/pytorch/executorch/pull/12285

Reviewed By: billmguo, cccclai

Differential Revision: D77910880
